test/Analysis/test-membar.mlir

```diff
@@ -611,7 +624,7 @@ tt.func @single_call_sync(%A : !tt.ptr<f16>) {
 tt.func @single_call_no_sync(%A : !tt.ptr<f16>) {
   // CHECK-NOT: gpu.barrier
   %0 = arith.constant dense<0.000000e+00> : tensor<16x16xf16, #AL>
-  tt.call @convert_layout2(%A) : (!tt.ptr<f16>) -> ()
+  tt.call @convert_layout5(%A) : (!tt.ptr<f16>) -> ()
   %1 = ttg.convert_layout %0 : tensor<16x16xf16, #AL> -> tensor<16x16xf16, #BL>
```

Comment:
The local_alloc calls in `convert_layout2` are 512 bytes, but this conversion uses more due to padding. Previously this still worked out because the allocations were being aligned to 1024 byte multiples, so the second alloc wouldn't interfere but now because we aren't over-aligning we do interfere. 

So, I've changed the test to call a function that creates a bigger allocation that it can safely reuse.